pkg/controller/sparkapplication/controller.go

```diff
@@ -55,6 +53,38 @@ const (
 
 var (
 	keyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
+
+	// Valid Driver State Transitions: map with valid pairs of oldState-> {newStates}
```

Comment:
I think it's worth putting code related to state transition validation into a separate go file, e.g., state_validation.go?

Reply:
sure, makes sense.